src/pages/Main/Main.js

```diff
@@ -1,17 +1,120 @@
 import React, { Component } from 'react';
+import CardLarge from './CardLarge/CardLarge';
+import Carousel from '../../components/Carousel/Carousel';
 import Nav from '../../components/Nav/Nav';
 import Footer from '../../components/Footer/Footer';
 import Card from '../../components/Card/Card';
+import CategoryImg from '../../components/CategoryImg/CategoryImg';
+import Slogan from './Slogan/Slogan';
 import './Main.scss';
 
 export default class Main extends Component {
+  constructor() {
+    super();
+
+    this.state = {
+      carouselImg: [],
+      sloganImg: [],
+      cardLarge: [],
+      index: 0,
+    };
+  }
+
+  componentDidMount() {
+    fetch('data/CarouselData.json')
+      .then(res => res.json())
+      .then(data => {
+        this.setState({
+          carouselImg: data.results.product_events,
+          cardLarge: data.results.category_events,
+          sloganImg: data.results.slogan_events,
+        });
+      });
+  }
+
+  handleClickPrev() {
+    if (this.state.index === 0) {
+      this.setState({
+        index: 0,
+      });
+    } else {
+      this.setState({
+        index: this.state.index - 1,
+      });
+    }
+
+    console.log('this prev', this.state.index);
+  }
+
+  handleClickNext() {
+    if (this.state.index === 3) {
+      this.setState({
+        index: 3,
+      });
+    } else {
+      this.setState({
+        index: this.state.index + 1,
+      });
+    }
+    console.log('this next', this.state.index);
+  }
+
   render() {
+    const { carouselImg, cardLarge, sloganImg } = this.state;
+
     return (
       <>
-        <div className="navWrapper">
-          <Nav />
-        </div>
-        <Card />
+        <Nav />
+        <main className="mainContainer">
+          <div className="abc">
+            <div className="slideBox">
+              <ul
+                className="slideFstWrapper"
+                style={{
+                  transform: `translateX(-${100 * this.state.index}vw)`,
+                }}
+              >
+                {carouselImg.map(img => (
+                  <Carousel key={img.product_id} img={img.image_url} />
+                ))}
+              </ul>
+            </div>
+
+            <div className="buttonWrapper">
+              <i
+                className="fas fa-chevron-left fa-4x"
+                onClick={() => this.handleClickPrev()}
+              ></i>
```

Comment:
```suggestion
              <i
                className="fas fa-chevron-left fa-4x"
                onClick={this.handleClickPrev}
              />
```
- 새로 함수 인스턴스를 만들지 않고 함수를 직접 넘겨주시면 될거 같아요

Reply:
넵 수정했습니다. 만약에 제가 원래 한 방식대로라면 인스턴스를 생성한 기록이 클릭할때마다 남아 있어서 비효율적이라 그런 걸까요??

Reply:
@Junghoon-P 오우 정답입니다! 기록이 남는다기 보다는 해당 부분이 실행될 때마다 기존에 생성해놓은 함수를 호출하는게 아니라 새로운 함수를 생성해서 실행하는 방식이라서 비효율적이라고 할 수 있습니다